### DISCUSSION TITLE
What API permissions for Remove-PnPSiteGroup?

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/powershell#687] [Q&A]
I need to use Remove-PnPSiteGroup as a registered application.

I connect with `$appConn = Connect-PnPOnline  -Url $siteURL -ClientId $ClientId `
but the line `Remove-PnPSiteGroup -Identity $GroupToRemove`
Gets an error: ` The remote server returned an error: (401) Unauthorized.`

this app has many Graph and SharePoint permissions.
What permissions are missing?
I intend to use it in unattended way, so it must be a registered application.

Thank you.

**Comment by AndersRask** [comment]
I just tested this with a service principal, and it seems to work fine.
Please specify the exact API permissions you have set on API? 

For SharePoint you would want something like Sites.FullControl 

If you could paste what you have under Manifest json > requiredResourceAccess
It should be something like
```
"requiredResourceAccess": [
		{
			"resourceAppId": "00000003-0000-0ff1-ce00-000000000000",
			"resourceAccess": [
				{
					"id": "678536fe-1083-478a-9c59-b99265e6b0d3",
					"type": "Role"
				}
			]
		}
	],
```
